src/js/tweet-parser.js

```diff
@@ -30,79 +92,238 @@ TweetParser.prototype = {
+    else {
+      text = tweet.querySelector(textSelector) ? tweet.querySelector(textSelector).textContent : null;
+    }
+  }
+  
+  return {
     username: user,
     text: text,
     id: tweetid,
-    links: getLinks(tweet),
     domObject: tweet,
-    analyzed: isAnalyzed 
+    analyzed: false,
+    logo: false
   };
 
-  return mock;
-
 };
 
-const indexTweets = (callback) => {
+const treatNewTweet = (tweet, num, callback) => {
+  const tweetInfo = getTweetInfo(tweet, null);
 
-  tweetsList.forEach(tweet => {
+  if (tweetInfo.username) {
 
-    const tweetInfo = getTweetInfo(tweet);
+    idAssignator(tweet);
+    tweet.setAttribute(usernameAttribute, tweetInfo.username);
+    callback(tweetInfo);
 
-    if (tweetInfo.username) {
+  }
+};
 
-      idAssignator(tweet);
-      tweet.setAttribute(usernameAttribute, tweetInfo.username);
-      callback(tweetInfo);
+const indexTweets = (callback) => {
 
-    }
+  tweetsList.forEach((tweet, num) => {
+
+    treatNewTweet(tweet, num, callback);
 
   });
 
 };
 
-const twitterMainChangeCallback = (change, callback) => {
-
-  if (!!change.querySelector(tweetJQuerySelector)) {
+const mainChangeCallback = (newNode, callback) => {
+
+  let auxTweetNode = null;
+  let auxSectionNode = null;
+  let newTweetNode = null;
+
+  // we have to check if there was a new tweet, or a new section
+  // to check it we have to consider whick user case we are in (logged / not logged)
+  if (userCase === null) {
+    try {
+      // we check if the new node is a new tweet itself
+      if (newNode.matches(tweetAlternativeSelector)) {
+        newTweetNode = newNode;
+      }
+      else {
+        // we check if the new node contains a new tweet
+        auxTweetNode = newNode.querySelector(tweetAlternativeSelector);
+        if (auxTweetNode) {
+          newTweetNode = auxTweetNode;
+          auxTweetNode = true;
+        }
+      }
+      // we check if the new node is a new section itself
+      auxSectionNode = newNode.matches(sectionAlternativeSelector);
+      if (!auxSectionNode) {
+        // we check if the new node contains a new section
+        auxSectionNode = newNode.querySelector(sectionAlternativeSelector);
+      }
+    } catch (e) {}
+  }
+  else {
+    try {
+      // we check if the new node is a new tweet itself
+      if (newNode.matches(tweetSelector)) {
+        newTweetNode = newNode;
+      }
+      else {
+        // we check if the new node contains a new tweet
+        auxTweetNode = newNode.querySelector(tweetSelector);
+        if (auxTweetNode) {
+          newTweetNode = auxTweetNode;
+          auxTweetNode = true;
+        }
+      }
+      // we check if the new node is a new section itself
+      auxSectionNode = newNode.matches(sectionSelector);
+      if (!auxSectionNode) {
+        // we check if the new node contains a new section
+        auxSectionNode = newNode.querySelector(sectionSelector);
+      }
+    } catch (e) {}
```

Comment:
Same comment as above

Reply:
Same answer as above. The issue is already created